hydromt/raster.py

```diff
@@ -2247,6 +2248,245 @@ def tile_window(shape, px):
+            Increase if memory issues are encountered.
+        driver : str, optional
+            GDAL driver (e.g., 'GTiff' for geotif files), or 'netcdf4' for netcdf files.
+        **kwargs
+            Key-word arguments to write raster files
+        """
+        resampling = getattr(Resampling, method, None)
+        if resampling is None:
+            raise ValueError(f"Resampling method unknown: {method}.")
+        px_size = 256  # Fixed resolution for osm
+
+        assert self._obj.ndim == 2, "Only 2d datasets are accepted..."
+        obj = self._obj.copy()
+        obj = obj.transpose(self.y_dim, self.x_dim)
+
+        mName = os.path.normpath(os.path.basename(root))
+
+        def create_folder(path):
+            if not os.path.exists(path):
+                os.makedirs(path)
+
+        if bbox:
+            minx, miny, maxx, maxy = bbox
+        else:
+            minx, miny, maxx, maxy = obj.raster.transform_bounds("EPSG:3857")
+
+        minx, miny = map(
+            max,
+            zip((minx, miny), [-20037508.34] * 2),
+        )
+        maxx, maxy = map(
+            min,
+            zip((maxx, maxy), [20037508.34] * 2),
+        )
+
+        _obj_tr = rasterio.warp.calculate_default_transform(
+            obj.raster.crs,
+            "EPSG:3857",
+            *obj.shape,
+            *obj.raster.bounds,
+        )[0]
+        dres = _obj_tr[0]
+        nzl = int(np.ceil((np.log10((20037508.34 * 2) / (dres * 256)) / np.log10(2))))
+
+        def tile_window(zl, minx, miny, maxx, maxy):
+            # Origin displacement
+            odx = np.floor(abs(-20037508.34 - minx) / dx)
+            ody = np.floor(abs(20037508.34 - maxy) / dx)
+
+            # Set the new origin
+            minx = -20037508.34 + odx * dx
+            maxy = 20037508.34 - ody * dx
+
+            # Create window generator
+            lu = product(
+                np.arange(round(minx, 2), round(maxx, 2), dx),
+                np.arange(round(maxy, 2), round(miny, 2), -dx),
+            )
+            for l, u in lu:
+                col = int(round(odx + (l - minx) / dx, 2))
+                row = int(round(ody + (maxy - u) / dx, 2))
+                bounds = [
+                    l,
+                    u - dx,
+                    l + dx,
+                    u,
+                ]
+                yield Affine(px, 0, l, 0, -px, u), bounds, col, row
+
+        zoom_levels.sort()
+        if zoom_levels[-1] > min_lvl:
+            min_lvl = zoom_levels[-1]
+        zoom_levels = list(range(zoom_levels[0], min_lvl + 1, 1))
+        zoom_levels.reverse()
+
+        create_folder(root)
+
+        vrt_fn = None
+        nodata = self.nodata
+        zls = {}
+        for zl in zoom_levels:
+            dx = (20037508.34 * 2) / (2**zl)
+            px = dx / px_size
+
+            if max(zoom_levels) > nzl:
+                logger.warning(
+                    f"Pixels at zoomlevel {zl} are smaller than the original data"
+                )
```

Comment:
when given this warning, would the user know what to do? or can we add some instructions here? (if the answer to the first question is yes, you can ignore this comment)

Reply:
Ill elaborate the warning at bit, consider it done.